### DISCUSSION TITLE
Replacing a file is confusing

### DISCUSSION THREAD

**Discussion by programmarchy** [directus/directus#19680] [Feature Requests]
### Summary

I can "Replace File" when editing from the File Library:

<img width="2032" alt="file-library" src="https://github.com/directus/directus/assets/622192/b613b432-e5a4-463a-9abe-358f9aeed595">

But I cannot "Replace File" from the detail pane opened from a File field:

<img width="2032" alt="field-details" src="https://github.com/directus/directus/assets/622192/11a9e7ff-9ba1-45e4-be8b-dbb9f6419282">

Because there is not a straightforward way to replace the stored file, it leads to orphaned files accumulating in the File Library.

This applies to File fields, in addition to Image fields.

### Motivation

The motivations are:

1. decreasing File Library clutter
2. improving navigation between items and related files

If a user wants to replace the stored file related to an item, it's confusing because there is not a way to do it directly from the item. Instead a user has to navigate to the File Library, search for the file they believe is related to the item, perform "Replace File", then navigate back to the item to verify their change.

### Detailed Design

I would propose two potential solutions:

1. Display the "Replace File" button (which exists in the File Library view) on the field details pane also.
3. Add a "Navigate to File" action on the field details pane to make it easy for a user to the correct file in the File Library, where they can perform the "Replace File" operation.

### Must Have:
- A straightforward way to replace the stored file of a File or Image related to an item.

### Should Have:
- N/A

### Could Have:
- N/A

### Won't Have:
- N/A

### Drawbacks

Maybe there is a reason the "Replace File" button wasn't included on the details pane?

Also, the iconography may be a little overloaded if the `launch` icon is used for "Navigate to File" as well as "Edit".

<img width="840" alt="navigate-to-item" src="https://github.com/directus/directus/assets/622192/ea18dc32-715f-4a84-9a61-f847e227091c">

<img width="415" alt="edit-image" src="https://github.com/directus/directus/assets/622192/e2508583-8990-4fad-80f1-bf1949c55c82">

### Alternatives

The impact of not doing this is a cluttered File Library and users being confused about how to navigate to files related to their items.

### Adoption Strategy

Should not be any breaking changes.

However, seems like non-technical users are not able to easily distinguish between replacing a "file record" (i.e. the row in `directus_files`) and "file data" (i.e. the actual file data in storage). Not sure how to improve the vocabulary or communicate the difference between the two.

**Comment by w0kyj** [comment]
Recommended Implementation:
File drawer should be reworked to use the file detail components so all functionality is implemented for both usages.

**Reply by programmarchy** [reply]
Dug into this and not sure if it's what you were referring to, but looks like `drawer-files.vue` is for picking a folder to place files. It's `drawer-items.vue` that appears when editing a File (or Image) field, which has a little extra handling for when the related item is a file. So that's where I put the "Replace File" button code in the PR linked below. Probably some improvements that can be made, but figure it makes sense to discuss those details in the PR.

**Comment by programmarchy** [comment]
Issue: https://github.com/directus/directus/issues/19730
PR: https://github.com/directus/directus/pull/19731
